libraries/chain/include/eosio/chain/abi_def.hpp

```diff
@@ -165,16 +214,80 @@ void from_variant(const fc::variant& v, eosio::chain::may_not_exist<T>& e) {
    from_variant( v, e.value );
 }
 
+
+template<typename ST, typename T>
+ST& operator << (ST& s, const eosio::chain::kv_tables_as_object<T>& v) {
+   raw::pack(s, v.value);
+   return s;
+}
+
+template<typename ST, typename T>
+ST& operator >> (ST& s, eosio::chain::kv_tables_as_object<T>& v) {
+   if (s.remaining())
```

Comment:
why are you making this object optional? can you point to the code where it is used?
few lines above this code there is similar method with `may_not_exist` template that is designed for such situations. so if you have optional object, what about wrapping it into `stream >> may_no_exist<object>` ?

Reply:
It is not optional.  In libraries/fc/include/fc/variant.hpp,  from_variant treats all var as an array. In kv_table it is varaint_object.

    454    template<typename K, typename T>
    455    void from_variant( const variant& var,  std::map<K, T>& vo )
    456    {
    457       const variants& vars = var.get_array();

Reply:
So if it is not optional, any reason to call ‘s.remaining()’?

Reply:
No, I will remove the remaining() check

Reply:
removing the remaining breaks existing unit tests

Reply:
so you may want to adjust tests. e.g. use `may_no_exist` or catch exception if it was expected

Reply:
When old abi which doesn't have kv_tables is passed and tried to read it gets an exception. The if(s.remaining()) is needed to backward compatibility with abi 1.0 and 1.1.
As it is not optional field in abi 1.2,  may_no_exist is not a good one to apply.

Reply:
I approve it now but suggest you to learn about SFINAE technique. that way you don't need this object at all.
with SFINAE you can define your `to_variant/from_variant` that way:
`template<typename T, typename = enable_if_t<is_same_v<T, your_object_type>, void>>
void from_variant(const fc::variant& v, eosio::chain::may_no_exist<T>& o) {}`